src/store/message.ts

```diff
@@ -61,6 +62,18 @@ export const useMessageStore = create<MessageState>((set) => ({
   },
   messageProfiles: new Map(),
   setMessageProfiles: (messageProfiles) => set(() => ({ messageProfiles })),
+  addProfileAndSelectTab: (key, profile) =>
+    set((state) => {
+      let profiles: Map<string, Profile>;
+      if (!state.messageProfiles.get(key)) {
```

Comment:
I tried to be smart here and only trigger a profile refresh if it's a new profile. This is also where we update the selected tab before opening the message component.